Expose macros config in babel

NullVoxPopuli opened the pull request:
@embroider/macros were not configurable in main/vite before this change.

Proposed way of configuring macros is similar to how we've been doing so in ember-cli-build.js

<details><summary>removed from this PR</summary>

## If we do end up needing this in the future,

we should have compat-macros output their combined options to compat-app-builder -> addBabelCompat

rather than the output,

and then have the babel plugin from compat read that in and it could then merge the options.

----------

For example, in the babel config:
```js
const {
  babelCompatSupport,
  templateCompatSupport,
} = require("@embroider/compat/babel");

module.exports = {
  plugins: [
   // ... 
    ...babelCompatSupport({
      '@embroider/macros': {
        setOwnConfig: {},
        setConfig: {
          'library-name': {}
        },
      }
    }),
  ],
  // ...
};
```

</details>

However, this doesn't solve the problem of configuring macros for the minimal app over in https://github.com/embroider-build/embroider/pull/2205

For the minimal app blueprint, we'd have folks do this in their babel config:

```js
const {
  babelCompatSupport,
  templateCompatSupport,
} = require("@embroider/compat/babel");

const { MacrosConfig } = require('@embroider/macros/src/macros-config');

let root = process.cwd();
let macros = MacrosConfig.for({}, root);
let { plugins: templateMacros, setConfig } = MacrosConfig.transforms();

setConfig(macros);

macros.setOwnConfig(root, { /* ... */ });
macros.setConfig(root, 'library-a', { /* ... */ });
macros.setConfig(root, 'library-b', { /* ... */ });

if (process.env.NODE_ENV === 'development') {
  macros.enablePackageDevelopment(process.cwd());
  macros.enableRuntimeMode();
}

macros.finalize();


module.exports = {
  plugins: [
    [
      "babel-plugin-ember-template-compilation",
      {
        compilerPath: "ember-source/dist/ember-template-compiler.js",
        transforms: [...templateMacros],
      },
    ],

   // ... 
   macros.babelPluginConfig();
  ],
  // ...
};
```
which.... could be better...

So.... I've done added to macros to allow for:
```js
const { buildMacros } = require('@embroider/macros/babel'); 

const macros = buildMacros({
  // this is how you configure your own package
  setOwnConfig: {
    // your config goes here
  },
  // this is how you can optionally send configuration into your
  // dependencies, if those dependencies choose to use
  // @embroider/macros configs.
  setConfig: {
    'some-dependency': {
      // config for some-dependency
    },
  },
});

module.exports = {
  plugins: [
   // ... 
    [
      "babel-plugin-ember-template-compilation",
      {
        compilerPath: "ember-source/dist/ember-template-compiler.js",
        transforms: [...macros.templateMacros],
      },
    ],
    ...macros.babelMacros,
  ],
  // ...
};

```

Which should be nice for the minimal blueprint in https://github.com/embroider-build/embroider/pull/2205

ef4 commented:
Notes from discussion at office hours:
 - we should make the argument to `babelCompatSupport` work even if you're using a prebuild, and we think that doesn't work now because when there's a prebuild the entire macros config gets built into _babel_compat_.js ahead of time and your options are ignored.

NullVoxPopuli commented:
I think in order to do that we need to have the classic macros config, in ember-cli-build.js, load the babel config.

It's too intense to re-apply the options post-macros-creation.

NullVoxPopuli commented:
err no, it'd just be the same problem, but in reverse

NullVoxPopuli commented:
Update: I removed the changes to `babelCompatSupport()`, and updated the description with notes from our meeting today for if/when we want to add in the merging

simonihmig commented on packages/macros/package.json:
```diff
@@ -14,6 +14,11 @@
+  "exports": {
+    ".": "./src/index.js",
+    "./babel": "./src/babel.js",
+    "./src/*": "./src/*.js"
```
Do we want people to import everything from `src`? Or is this for backwards compatibility, because people were doing so before?

NullVoxPopuli replied:
its backwards compatibility because _we_ were doing it before.
I didn't audit what all we're importing, but it probably would be good to restrict at some point.

mansona replied:
if we're going to restrict it now is the time. We won't be able to reduce it once this is landed because we really don't want a new major. Can we just remove it and see what fails?

NullVoxPopuli replied:
I'm more than happy to do a follow-up PR which removes `src`.
I don't really want to make this PR do 2 things (1. current focus, 2. change available modules)

simonihmig replied:
I agree with both of you: let's remove it, but in a follow-up PR :)

simonihmig reviewed (approved):
LGTM, but I'll leave the final call to more knowledgeable people! 🙃
